rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/InfrastructureManager.java

```diff
@@ -769,7 +769,10 @@ protected final String addDeployingNode(String name, String command, String desc
             // This is due to deep copies made by ProActive Programming with method invocation on Active Objects.
             // As a consequence, the 'deployingNode' variable must be updated with the last value available
             // in the 'deployingNodes' collection
-            deployingNode = getDeployingOrLostNode(deployingNodeUrl);
+            RMDeployingNode statefulDeployingNode = getDeployingOrLostNode(deployingNodeUrl);
+            if (statefulDeployingNode != null) {
+                deployingNode = statefulDeployingNode;
+            }
```

Comment:
What about using Optional?

Reply:
Since I am doing an assignment here (side effect), I cannot easily use an optional. Indeed inside a lambda, only side effect-free expressions are allowed.